numpy/_core/src/multiarray/convert_datatype.c

```diff
@@ -2749,7 +2749,7 @@ nonstructured_to_structured_resolve_descriptors(
 
             Py_ssize_t pos = 0;
             PyObject *key, *tuple;
-            while (PyDict_Next(to_descr->fields, &pos, &key, &tuple)) {
+            while (PyDict_Next(to_descr->fields, &pos, &key, &tuple)) { // noqa: borrowed-ref - manual fix needed
```

Comment:
fields again, this is fine, ditto for all the other ones that go through PyDataType_FIELDS below.